### PR TITLE
Move ConsoleIntegrationTestTrait to console package

### PR DESCRIPTION
**Pull request by othercorey**
This moves the trait needed to test console into the console package along with dependencies. Users should be able to test code using split packages without requiring the full cakephp/cakephp monolithic package. Without this, users aren't able to test with the same dependencies used to deploy/release.

Aliases were created in the TestSuite package.

### CODE REVIEW

**Comment by othercorey** [comment]
This is now the use statement list in `ConsoleIntegrationTestTrait` with 0 references to `Cake\TestSuite`:

```php
use Cake\Command\Command;
use Cake\Console\CommandRunner;
use Cake\Console\ConsoleIo;
use Cake\Console\Exception\StopException;
use Cake\Console\TestSuite\Constraint\ContentsContain;
use Cake\Console\TestSuite\Constraint\ContentsContainRow;
use Cake\Console\TestSuite\Constraint\ContentsEmpty;
use Cake\Console\TestSuite\Constraint\ContentsNotContain;
use Cake\Console\TestSuite\Constraint\ContentsRegExp;
use Cake\Console\TestSuite\Constraint\ExitCode;
use Cake\Core\TestSuite\ContainerStubTrait;
use RuntimeException;
```

**Review by markstory — COMMENTED** [review]
Looks good to me.

File: src/Console/TestSuite/ConsoleIntegrationTestTrait.php [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+        try {
+            $this->_exitCode = $runner->run($args, $io);
+        } catch (MissingConsoleInputException $e) {
+            $messages = $this->_out->messages();
+            if (count($messages)) {
+                $e->setQuestion($messages[count($messages) - 1]);
+            }
+            throw $e;
+        } catch (StopException $exception) {
+            $this->_exitCode = $exception->getCode();
```
Comment (othercorey) [inline src/Console/TestSuite/ConsoleIntegrationTestTrait.php:114]: @markstory should we be testing this?

Comment (markstory) [reply src/Console/TestSuite/ConsoleIntegrationTestTrait.php:114]: Probably. I thought we would have had a test already 🤔

**Pull request merged**